sweepai/utils/openai_proxy.py

```diff
@@ -0,0 +1,28 @@
+import os
+import openai
+
+class OpenAIProxy:
+    def __init__(self, OPENAI_API_KEY_GPT35, OPENAI_API_KEY_GPT4, OPENAI_API_KEY_GPT4_32K, OPENAI_FALLBACK):
+        self.OPENAI_API_KEY_GPT35 = OPENAI_API_KEY_GPT35
+        self.OPENAI_API_KEY_GPT4 = OPENAI_API_KEY_GPT4
+        self.OPENAI_API_KEY_GPT4_32K = OPENAI_API_KEY_GPT4_32K
+        self.OPENAI_FALLBACK = OPENAI_FALLBACK
+
+    def call_openai(self, model, **kwargs):
+        try:
+            if model == 'gpt3.5':
```

Comment:
make sure to route the call to azure this way:
import this:
```
from sweepai.config.server import OPENAI_API_TYPE
```
for each request check if OPENAI_API_TYPE == "azure" and whether OPENAI_API_KEY_<MODEL> is not none
fallback is at the model level, not as it's shown here. we should first try azure then try openai, instead of trying openai and then trying 3.5
```
                        output = (
                            openai.ChatCompletion.create(
                                engine=OPENAI_API_ENGINE
                                if OPENAI_API_TYPE == "azure"
                                else None,
                                model=model,
                                messages=messages_dicts,
                                max_tokens=max_tokens - token_sub,
                                temperature=temperature,
                                functions=[
                                    json.loads(function.json())
                                    for function in functions
                                ],
                                function_call=function_name,
                            )
                            .choices[0]
                            .message                       
```